### PR TITLE
Add an error check when the user try to add a note for a non-existing person record. #221

### PR DESCRIPTION
**Pull request by gimite**

### CODE REVIEW

File: tests/scrape.py [hunk 1]
```diff
@@ -170,8 +170,17 @@ def fetch(url, data='', agent=None, referrer=None, charset=None, verbose=0,
...
-        data = urlencode(data)
+
+    if not data:
+        data_str = ''
+    elif isinstance(data, str):
+        data_str = data
+    elif isinstance(data, unicode):
+        data_str = data.encode('utf-8')
+    elif isinstance(data, dict):
+        data_str = urlencode(data)
```
Comment (skywhale) [inline tests/scrape.py]: could it also be a dict of unicodes? I haven't followed through the code.

File: app/view.py [hunk 2]
```diff
@@ -182,6 +182,9 @@ def post(self):
                        '"believed_dead".'))
 
         person = Person.get(self.repo, self.params.id)
+        if not person:
```
Comment (skywhale) [inline app/view.py]: Nice! It's a bit surprising that he weren't checking the condition.

**Comment by skywhale** [comment]
LGTM. Thanks for the bug fix!

File: tests/scrape.py (hunk 1 above)
Comment (gimite) [inline tests/scrape.py]: Reading the code, urlquote(), used in urlencode(), supports both str and unicode. Added a comment.

**Comment by gimite** [comment]
Thanks. Merged.

**Pull request merged**